packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -377,15 +377,15 @@ function focusFirstInScope(scope: HTMLElement[]) {
 
 function useAutoFocus(scopeRef: RefObject<HTMLElement[]>, autoFocus: boolean) {
   const autoFocusRef = React.useRef(autoFocus);
-  useEffect(() => {
+  useLayoutEffect(() => {
     if (autoFocusRef.current) {
       activeScope = scopeRef;
       if (!isElementInScope(document.activeElement, activeScope.current)) {
         focusFirstInScope(scopeRef.current);
       }
     }
     autoFocusRef.current = false;
-  }, []);
+  });
```

Comment:
run on every render? why not just on mount? that autoFocusRef will ensure the body of this layouteffect is only run once anyways, and it'll be on mount

Reply:
I'll add ` // eslint-disable-next-line react-hooks/exhaustive-deps`.